napari/layers/utils/text.py

```diff
@@ -60,44 +75,32 @@ class TextManager:
         Set to true of the text should be displayed.
     """
 
-    def __init__(
-        self,
-        text,
-        n_text,
-        properties={},
-        rotation=0,
-        translation=0,
-        anchor='center',
-        color='cyan',
-        size=12,
-        blending='translucent',
-        visible=True,
-    ):
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            text=Event,
-            rotation=Event,
-            translation=Event,
-            anchor=Event,
-            color=Event,
-            size=Event,
-            blending=Event,
-            visible=Event,
-        )
-
-        self.events.block_all()
-        self._rotation = rotation
-        self._anchor = Anchor(anchor)
-        self._translation = translation
-        self._color = transform_color(color)[0]
-        self._size = size
-        self._blending = self._check_blending_mode(blending)
-        self._visible = visible
-
-        self._set_text(text, n_text, properties)
-        self.events.unblock_all()
+    text: InitVar[Union[np.ndarray, str]]
+    n_text: InitVar[int]
+    properties: InitVar[Dict[str, np.ndarray]] = None
+
+    rotation: float = 0.0
+    translation: Property[np.ndarray, None, np.asarray] = np.asarray(0)
+    anchor: Property[Anchor, str, Anchor] = Anchor.CENTER
+    color: Property[
+        Union[np.ndarray, str], None, transform_single_color
+    ] = 'cyan'
```

Comment:
Can we force this to be an array? It will only be string if people use default, otherwise will always be array